opal/browser/audio/node.rb

```diff
@@ -0,0 +1,149 @@
+module Browser; module Audio; module Node;
+
+  module Pluggable
+    def connect(other_node)
+      `#@native.connect(#{Native.convert(other_node)})`
+    end
+
+    def disconnect(other_node, options = {})
+      other_node = Native.try_convert(other_node)
+      output     = options[:output] || 0
+      input      = options[:input]  || 0
+
+      if options
+        `#@native.disconnect(#{other_node}, #{output}, #{input}) || nil`
+      else
+        if other_node
+          `#@native.disconnect(#{other_node})`
+        else
+          `#@native.disconnect()`
+        end
+      end
+    end
+  end
+
+  class Gain
+    include Native
+    include Pluggable
+
+    alias_native :value, :gain
+
+    def initialize(audio_context)
+      super `#{audio_context.to_n}.createGain()`
+    end
+  end
+
+  class Oscillator
+    TYPES = %i(sine square sawtooth triangle custom)
+
+    include Native
+    include Pluggable
+
+    attr_reader :frequency, :detune, :type
+
+    alias_native :start
+    alias_native :stop
+
+    def initialize(audio_context)
+      super `#{audio_context.to_n}.createOscillator()`
+    end
+
+    def type=(type)
+      unless TYPES.include?(type)
+        raise ArgumentError, "type #{type} doesn't exists"
+      end
+      `#@native.type = #{@type = type}`
+    end
+
+    def frequency=(frequency)
+      `#@native.frequency.value = #{@frequency = frequency}`
+    end
+
+    def detune=(detune)
+      `#@native.detune.value = #{@detune = detune}`
+    end
+  end
+
+  class Delay
+    include Native
+    include Pluggable
+
+    attr_reader :time
+
+    def initialize(audio_context, max_time)
+      super `#{audio_context.to_n}.createDelay(max_time || 1)`
+    end
+
+    def time=(time)
```

Comment:
@meh you mean if i add an `alias_native :time` i should get access to the setter `value` method ??